feat: Add switchToNextScreen hotkey to cycle through computers

KrisEnigma opened the pull request:
- Add switchToNextScreen action type and GUI option
- Implement server-side screen cycling logic
- Update configuration parsing for new action
- Fix Action::text() method indexing bug
- Update copyright headers for 2025

This implements GitHub issue https://github.com/deskflow/deskflow/issues/8006 - a hotkey to cycle through
connected computers in the Deskflow network.

Closes https://github.com/deskflow/deskflow/issues/8006

sithlord48 commented on src/lib/gui/Action.cpp:
```diff
@@ -12,7 +13,7 @@
 QString Action::text() const
 {
-  auto text = QString(m_actionTypeNames.at(keySequence().isMouseButton() ? type() + 6 : type()));
+  auto text = QString(m_actionTypeNames.at(type()));
```
Not sure what this "Fixes" but you should split it into its own commit.

KrisEnigma replied:
Hey! That isolated commit fixes a bug where mouse button actions were using type() + 6 instead of just type() for array indexing, causing out-of-bounds crashes. Split it out because it's a separate fix from the main feature - didn't want to hide it in the big commit!

sithlord48 replied:
@nbolton do you recall why we may have done this ?

nbolton replied:
This is Volker's old code. I'd say ask him, but unfortunately he [died](https://kde.org/announcements/4/4.14.0/) :cry:

sithlord48 reviewed (changes requested):
Overall looks ok need to split your fix into its own commit.

sithlord48 reviewed (changes requested):
Just some things: 

You should start the commits message with `fix:` and `feat:` (must be lowercase and have the `:`)

The commits here are not in order and are broken up incorrectly (for us) 

Lets get this commits fixed you have two commits here  (in this order if you don't mind) 

1. fixes the mouse label issue  (changes to Action.cpp) 
2. The rest that adds the feature (documentation etc is added in the same commit)

I have not tested the code yet by running it.

KrisEnigma reviewed (commented):
Added 2 commits as requested.

sithlord48 commented on src/lib/server/Server.cpp:
```diff
@@ -1330,6 +1333,41 @@ void Server::handleSwitchInDirectionEvent(const Event &event)
+
+  // Find the current active screen
+  std::string currentScreen = getName(m_active);
+  auto it = std::find(screens.begin(), screens.end(), currentScreen);
```
This does not build for me 
To fix i locally did `auto it = std::ranges::find(screens, currentScreen)` 

Try it with ranges and see if that works for you if so lets use it. If not include the header for `std::find`

KrisEnigma commented on src/lib/gui/Action.cpp:
```diff
@@ -1,5 +1,6 @@
 /*
  * Deskflow -- mouse and keyboard sharing utility
+ * SPDX-FileCopyrightText: (C) 2025 Deskflow Developers
```
Ok, I moved it!

KrisEnigma commented:
Hi @sithlord48 and @nbolton, I've addressed the feedback:
- Moved the copyright header to the first commit.
- Fixed the std::find issue with std::ranges::find.
- Kept the commits in the fix:/feat: format.
I've pushed the updates and requested reviews. Let me know if there's anything else!

KrisEnigma commented on src/lib/server/Server.cpp:
```diff
@@ -1330,6 +1334,41 @@ void Server::handleSwitchInDirectionEvent(const Event &event)
+  // Find the client for the next screen
+  ClientList::const_iterator clientIt = m_clients.find(*nextIt);
+  if (clientIt == m_clients.end()) {
+    LOG_DEBUG1("next screen not active");
```
Got it! I've changed the logging levels to LOG_ERR and pushed the changes.
